build.gradle

```diff
@@ -535,6 +537,76 @@ configure(subprojects - getJsProjects()) { subproject ->
     }
 }
 
+def dexerProjects = getAndroidProjects().findAll {
+    !it.name.contains("-specs-")
+}
+configure(dexerProjects) { subproject ->
+    configurations {
+        r8
+        d8
+    }
+    repositories{
+        google()
+    }
+    dependencies {
+        r8 "com.android.tools:r8:$d8_version"
+        d8 subproject
+    }
+
+    afterEvaluate {
+        def checkDexer = task('checkDexer', type: JavaExec, description: 'compiles android class files into dex bytecode', dependsOn: jar) {
+            outputs.dir("${subproject.buildDir}/kotlin/d8")
+
+            classpath subproject.configurations.r8.asPath
+            main = "com.android.tools.r8.D8"
+
+            def d8Classpath = subproject.configurations.d8.files.collect {
+                ["--classpath", it.getAbsolutePath()]
+            }.flatten()
+
+            args = d8Classpath + [
+                // ATRIUM_ANDROID_JAR has to point to something like android-sdk-linux/platforms/android-XY/android.jar where Xy is the version we want to support
+                '--lib', System.getenv('ATRIUM_ANDROID_JAR'),
+                '--output', "${subproject.buildDir}/kotlin/d8",
+                jar.archiveFile.get().asFile
+            ]
+        }
```

Comment:
you need to include a check here that we did not have warnings in the output. I think you could do it with setting
```
standardOutput = yourOwnOutputStream
errorOutput = yourOwnOutputStream
```
and then perform a check in `doLast`